### PR TITLE
feat: add hires fix option for sd-webui

### PR DESCRIPTION
**Pull request by MaikoTan**
Par https://github.com/koishijs/novelai-bot/issues/64#issuecomment-1295863037

La fonctionnalité de "face restoration" est compliquée, elle serait implémentée dans le prochain PR.

------

根据 https://github.com/koishijs/novelai-bot/issues/64#issuecomment-1295863037

“人脸修复”功能是复杂的，她会在下一个PR中被实现。

### CODE REVIEW

**Review by shigma — COMMENTED** [review]
这个不默认开启是有什么考虑吗？

**Comment by MaikoTan** [comment]
the "hires fix" function is actually:
- generate a small image first
- then img2img it to the request resolution

while it performs well when you're trying to generate larger images like 1024*1024, but in most cases like 512*768 or something, this function would not help much, but break the reproduction of the same prompt.
So I don't think it is worth to be enabled by default.

**Comment by medicago087** [comment]
Because the training model of one person is usually 512*512~768*768, according to my experience, I think we can consider setting a switch to enable this function automatically when the length and width of image are both higher than 1024.

Of course, it is also feasible to just identify it as an optional parameter.

**Comment by MaikoTan** [comment]
OK I have resolved the conflict that occurred on the updated main branch.

File: src/index.ts [hunk 1]
```diff
@@ -107,6 +107,7 @@ export function apply(ctx: Context, config: Config) {
     .option('scale', '-c <scale:number>')
     .option('noise', '-n <noise:number>', { hidden: some(restricted, thirdParty) })
     .option('strength', '-N <strength:number>', { hidden: restricted })
+    .option('hiresfix', '-H', { hidden: () => config.type !== 'sd-webui' })
```
Comment (shigma) [inline src/index.ts:110]: Would `hiresFix` be better?

Comment (MaikoTan) [inline src/index.ts:110]:
```suggestion
    .option('hiresFix', '-H', { hidden: () => config.type !== 'sd-webui' })
```

File: src/index.ts [hunk 2]
```diff
@@ -264,6 +265,7 @@ export function apply(ctx: Context, config: Config) {
         return {
           sampler_index: sampler.sd[options.sampler],
           init_images: image && [image.dataUrl], // sd-webui accepts data URLs with base64 encoded image
+          enable_hr: options.hiresfix ?? config.hiresFix ?? false,
```
Comment (MaikoTan) [inline src/index.ts:268]:
```suggestion
          enable_hr: options.hiresFix ?? config.hiresFix ?? false,
```

File: src/index.ts (hunk 1 above)
Comment (MaikoTan) [reply src/index.ts]: Done !

File: src/locales/zh-CN.yml [hunk 3]
```diff
@@ -17,6 +17,7 @@ commands:
       scale: 设置对输入的服从度
       strength: 图片修改幅度
       noise: 图片噪声强度
+      hiresfix: 启用高分辨率修复
```
Comment (shigma) [inline src/locales/zh-CN.yml:20]:
```suggestion
      hiresFix: 启用高分辨率修复
```

**Review by shigma — APPROVED** [review]

**Pull request merged**